Shared/Samples/Add custom dynamic entity data source/AddCustomDynamicEntityDataSourceView.swift

```diff
@@ -0,0 +1,164 @@
+    
+    /// The point on the screen the user tapped.
+    @State private var tappedScreenPoint: CGPoint?
+    
+    /// The placement of the callout.
+    @State private var calloutPlacement: CalloutPlacement?
+    
+    init() {
+        // The meta data for the custom dynamic entity data source.
+        let info = DynamicEntityDataSourceInfo(
+            entityIDFieldName: "MMSI",
+            fields: .vesselFields
+        )
+        
+        info.spatialReference = .wgs84
+        
+        let customDataSource = CustomDynamicEntityDataSource(info: info) { VesselFeed() }
+        
+        _dynamicEntityLayer = .init(initialValue: DynamicEntityLayer(dataSource: customDataSource))
+        
+        let trackDisplayProperties = dynamicEntityLayer.trackDisplayProperties
+        trackDisplayProperties.showsPreviousObservations = true
+        trackDisplayProperties.showsTrackLine = true
+        trackDisplayProperties.maximumObservations = 20
+        
+        let labelDefinition = LabelDefinition(
+            labelExpression: SimpleLabelExpression(simpleExpression: "[VesselName]"),
+            textSymbol: TextSymbol(color: .red, size: 12)
+        )
+        labelDefinition.placement = .pointAboveCenter
+        
+        dynamicEntityLayer.addLabelDefinition(labelDefinition)
+        dynamicEntityLayer.labelsAreEnabled = true
+        
+        map.addOperationalLayer(dynamicEntityLayer)
+    }
+    
+    var body: some View {
+        MapViewReader { proxy in
+            MapView(map: map)
+                .onSingleTapGesture { screenPoint, _ in
+                    tappedScreenPoint = screenPoint
+                }
+                .callout(placement: $calloutPlacement.animation(.default.speed(2))) { placement in
+                    let attributes = (placement.geoElement as! DynamicEntityObservation).attributes
+                    VStack(alignment: .leading) {
+                        // Display all the attributes in the callout.
+                        ForEach(attributes.sorted(by: { $0.key < $1.key }), id: \.key) { item in
+                            Text("\(item.key): \(String(describing: item.value))")
+                        }
+                    }
+                }
+                .task(id: tappedScreenPoint) {
+                    guard let tappedScreenPoint,
+                          let identifyResult = try? await proxy.identify(
+                            on: dynamicEntityLayer,
+                            screenPoint: tappedScreenPoint,
+                            tolerance: 2
+                          ) else {
+                        // Hides the callout.
+                        calloutPlacement = nil
+                        
+                        return
+                    }
+                    
+                    // Set the callout placement to the observation that was tapped on.
+                    calloutPlacement = identifyResult.geoElements.first.map { .geoElement($0) }
```

Comment:
Never use a guard statement when an if…else will do. Else clauses of guard statements shouldn't do any work. Since forward progress is being made in this case, we should use an if…else and enforce that `calloutPlacement` is always updated.

```suggestion
                    let newCalloutPlacement: CalloutPlacement?
                    if let tappedScreenPoint,
                       let identifyResult = try? await proxy.identify(
                        on: dynamicEntityLayer,
                        screenPoint: tappedScreenPoint,
                        tolerance: 2
                       ) {
                        // Set the callout placement to the observation that was tapped on.
                        newCalloutPlacement = identifyResult.geoElements.first.map { .geoElement($0) }
                    } else {
                        // Hides the callout.
                        newCalloutPlacement = nil
                    }
                    calloutPlacement = newCalloutPlacement
```